### ISSUE TITLE
[backport v1.2.2][BUG] pcidevices device plugin fails when device name is too long

### ISSUE THREAD

**Issue by khushboo-rancher** [harvester/harvester#5359]
**Describe the bug**
Backport of https://github.com/harvester/harvester/issues/4806

**Comment by ibrokethecloud** [comment]
backport available in pcidevices v0.2.7

**Comment by ibrokethecloud** [comment]
fix is now available in v1.2-head

**Comment by irishgordo** [comment]
This looks good on v1.2.2-rc2, as:
```
testv122rc2node5:/home/rancher # ls -alh /var/lib/kubelet/device-plugins/
total 44K
drwxr-xr-x 2 root root 4.0K May  7 19:06 .
drwxr-xr-x 8 root root 4.0K May  7 18:50 ..
srwxr-xr-x 1 root root    0 May  7 18:55 kubelet.sock
-rw------- 1 root root  33K May  7 19:06 kubelet_internal_checkpoint
srwxr-xr-x 1 root root    0 May  7 18:58 kubevirt-kvm.sock
srwxr-xr-x 1 root root    0 May  7 19:06 kubevirt-realtek.com-8168.sock
srwxr-xr-x 1 root root    0 May  7 18:58 kubevirt-tun.sock
srwxr-xr-x 1 root root    0 May  7 18:58 kubevirt-vhost-net.sock
```

When previously with v1.2.1, error is shown, on activating the pcid device as the name is too long previously, v1.2.1:

https://github.com/harvester/harvester/assets/5370752/6ac94567-d522-4454-95db-6de09719b5db

Whereas, now w/ v1.2.2-rc2, that error isn't present in the logs of the pcidevices controller pod and it doesn't have any problem enabling that device even if it's name is very long:
```
time="2024-05-07T19:06:45Z" level=info msg="Adding testv122rc2node5-000008000 to KubeVirt list of permitted devices"
W0507 19:06:45.523193       1 warnings.go:70] kubevirt.io/v1alpha3 is now deprecated and will be removed in a future release.
W0507 19:06:45.550370       1 warnings.go:70] kubevirt.io/v1alpha3 is now deprecated and will be removed in a future release.
time="2024-05-07T19:06:45Z" level=info msg="Enabling passthrough for PDC: testv122rc2node5-000008000"
time="2024-05-07T19:06:45Z" level=info msg="Binding device testv122rc2node5-000008000 [10ec 8168] to vfio-pci"
time="2024-05-07T19:06:45Z" level=info msg="Binding device 0000:08:00.0 vfio-pci"
time="2024-05-07T19:06:45Z" level=error msg="error syncing 'testv122rc2node5-000008000': handler PCIDeviceClaimReconcile: error writing to bind file: write /sys/bus/pci/drivers/vfio-pci/bind: device or resource busy, requeuing"
time="2024-05-07T19:06:45Z" level=info msg="[PCIDeviceController] Device testv122rc2node5-0000ff140 does not exist"
time="2024-05-07T19:06:45Z" level=info msg="Creating PCI Device: testv122rc2node5-0000ff140\n"
time="2024-05-07T19:06:46Z" level=info msg="Adding testv122rc2node5-000008000 to KubeVirt list of permitted devices"
W0507 19:06:46.324386       1 warnings.go:70] kubevirt.io/v1alpha3 is now deprecated and will be removed in a future release.
time="2024-05-07T19:06:46Z" level=info msg="Creating DevicePlugin: realtek.com/8168"
time="2024-05-07T19:06:46Z" level=info msg="Started DevicePlugin: realtek.com/8168"
{"component":"","level":"info","msg":"Initialized DevicePlugin: realtek.com/8168","pos":"device_manager.go:206","timestamp":"2024-05-07T19:06:46.327214Z"}
time="2024-05-07T19:06:47Z" level=info msg="Adding testv122rc2node5-000008000 to KubeVirt list of permitted devices"
W0507 19:06:47.122026       1 warnings.go:70] kubevirt.io/v1alpha3 is now deprecated and will be removed in a future release.
time="2024-05-07T19:06:47Z" level=info msg="[PCIDeviceController] Device testv122rc2node5-0000ff141 does not exist"
time="2024-05-07T19:06:47Z" level=info msg="Creating PCI Device: testv122rc2node5-0000ff141\n"
time="2024-05-07T19:06:47Z" level=info msg="Adding testv122rc2node5-000008000 to KubeVirt list of permitted devices"
W0507 19:06:47.722448       1 warnings.go:70] kubevirt.io/v1alpha3 is now deprecated and will be removed in a future release.
time="2024-05-07T19:06:48Z" level=info msg="[PCIDeviceController] Device testv122rc2node5-0000ff142 does not exist"
time="2024-05-07T19:06:48Z" level=info msg="Creating PCI Device: testv122rc2node5-0000ff142\n"

```
